test/multipart_test.gleam

```diff
@@ -0,0 +1,83 @@
+import bucket/complete_multipart_upload
+import bucket/create_multipart_upload
+import bucket/get_object
+import bucket/upload_part
+import gleam/bit_array
+import gleam/httpc
+import gleam/list
+import gleam/string
+import gleeunit/should
+import helpers
+
+pub fn perform_multipart_upload_test() {
+  // Set up a test bucket
+  helpers.delete_existing_buckets()
+  let bucket = "multipart1"
+  helpers.create_bucket(bucket)
+
+  // Initiate a multipart upload
+  let key = "test/myfile"
+  let assert Ok(res) =
+    create_multipart_upload.request(bucket:, key:)
+    |> create_multipart_upload.build(helpers.creds)
+    |> httpc.send_bits
+  let assert Ok(res) = create_multipart_upload.response(res)
+  should.equal(res.bucket, bucket)
+  should.equal(res.key, key)
+  should.not_equal(res.upload_id, "")
+
+  // Upload some parts (they can be sent in parallel)
+  // NOTE: The minimum Part size for multipart upload is 5MiB (hardcoded in minio)
+  let upload_id = res.upload_id
+  let part_size = 5 * 1024 * 1024
+  let part1 = get_random_bytes(part_size)
+  let part2 = get_random_bytes(part_size)
+  let part3 = <<"Goodbye!":utf8>>
+  let parts =
+    [part1, part2, part3]
+    |> list.index_map(fn(body, i) {
+      let part_number = i + 1
+      let assert Ok(res) =
+        upload_part.request(bucket:, key:, upload_id:, part_number:, body:)
+        |> upload_part.build(helpers.creds)
+        |> httpc.send_bits
+      let assert Ok(res) = upload_part.response(res)
+      should.not_equal(res.etag, "")
+      complete_multipart_upload.Part(part_number:, etag: res.etag)
+    })
+
+  // Complete the multipart upload
+  // NOTE: The Parts must be sorted in ascending order for this operation
+  let assert Ok(res) =
+    complete_multipart_upload.request(bucket:, key:, upload_id:, parts:)
+    |> complete_multipart_upload.build(helpers.creds)
+    |> httpc.send_bits
+  let assert Ok(res) = complete_multipart_upload.response(res)
+  should.be_true(string.contains(res.location, "/" <> bucket <> "/" <> key))
+  should.equal(res.bucket, bucket)
+  should.equal(res.key, key)
+  should.not_equal(res.etag, "")
+
+  // Get the uploaded object and verify its contents
+  let assert Ok(res) =
+    get_object.request(bucket, key)
+    |> get_object.build(helpers.creds)
+    |> httpc.send_bits
+  let assert Ok(get_object.Found(contents)) = get_object.response(res)
```

Comment:
Move this to a function in helpers.gleam please

Reply:
I'm not sure what you mean. Should I make a helper that takes a bucket, key, creds, and a list of parts as BitArrays, and verifies the object?

Reply:
Yeah! `helper.get_object` or something which returns the contents

Reply:
done in 57a99e4